lib/rubocop/cop/bundler/gem_version.rb

```diff
@@ -42,7 +42,9 @@ class GemVersion < Base
 
         # @!method includes_version_specification?(node)
         def_node_matcher :includes_version_specification?, <<~PATTERN
-          (send nil? :gem <(str #version_specification?) ...>)
+          (send nil? :gem
+            <{(str #version_specification?)
+              (hash <(pair (sym :tag) (str _)) (pair (sym {:git :github}) (str _)) ...>)} ...>)
```

Comment:
Is matching `git` or `github` required? Would `(sym :tag)` be enough to indicate tag presence? That way the cop doesn't need to be concerned about the gem source.

If it is needed, it looks like [`bitbucket`](https://bundler.io/v2.2/man/gemfile.5.html#BITBUCKET) is also a valid option to provide.

Reply:
I did not want this to report false positives - hence the extra check. 

Thanks I was not aware of the `bitbucket` option. Looks like there is `gist` as well. I'll add them both.

Reply:
Added `bitbucket` only, not sure if `gist` allows versions.

Reply:
Users can also add their own arbitrary git sources using `git_source` so we should probably handle that as well.

Reply:
@dvandersluis I think this might be tricky given that we can define custom sources:

```ruby
git_source(:stash) { |repo_name| "https://stash.corp.acme.pl/#{repo_name}.git" }
...
gem 'rails', :stash => 'forks/rails'
```

Reply:
I agree it's tricky, but if we don't handle it someway it leads to false reporting (which is maybe another argument for just looking for `tag`/`ref`/etc. and not caring about the other keys).